addons/sale_timesheet/models/sale_order.py

```diff
@@ -17,6 +17,10 @@ class SaleOrder(models.Model):
     tasks_ids = fields.Many2many('project.task', compute='_compute_tasks_ids', string='Tasks associated to this sale')
     tasks_count = fields.Integer(string='Tasks', compute='_compute_tasks_ids', groups="project.group_project_user")
 
+    visible_project = fields.Boolean('Display project id?', compute='_compute_visible_project', readonly=True)
+    project_id = fields.Many2one(
```

Comment:
Add `readonly=True, states={'draft': [('readonly', False)], 'sent': [('readonly', False)]}` to ensure it is readonly when SO is confirmed or cancelled.

Reply:
Nice mechanism, didn't know it. I've added the code.